src/Threads.cpp

```diff
@@ -134,6 +131,9 @@ void CThreadPool::threadFunc() {
 	assert(m_num_threads > 0 && m_num_idle > 0);
 	m_num_threads--;
 	m_num_idle--;
+
+	if (m_num_threads == 0)
```

Comment:
I'm not sure how this will interact with MAX_IDLE_THREADS... Let's check for m_done here.